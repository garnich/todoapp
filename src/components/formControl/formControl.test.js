import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FormControl from './formControl'

Enzyme.configure({ adapter: new Adapter() })

describe('Test FormControl component', () => {
  const props = {
    searchParam: jest.fn(),
  }

  it('<FormControl>', () => {
    const wrapper = shallow(<FormControl {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('<FormControl> test changeSearch', () => {
    const wrapper = shallow(<FormControl {...props} />)
    wrapper
      .find('.form-control')
      .simulate('change', { target: { value: 'newItem' } })

    expect(props.searchParam).toHaveBeenCalled()
  })
})
