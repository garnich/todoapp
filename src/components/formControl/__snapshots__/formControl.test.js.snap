// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FormControl component <FormControl> 1`] = `
<div
  className="search"
>
  <input
    className="form-control"
    onChange={[Function]}
  />
</div>
`;
